eslintrc.js

```diff
@@ -20,12 +20,34 @@ module.exports = {
     "sort-requires"
   ],
   "rules": {
+    "array-callback-return": "error",
+    "dot-notation": "error",
     "eqeqeq": "error",
+    "eslint-comments/no-unused-disable": "error",
+    "func-name-matching": "error",
+    "guard-for-in": "error",
     "no-console": "error",
+    "no-else-return": [
+      "error",
+      {
+        "allowElseIf": false
+      }
+    ],
+    "no-implicit-coercion": "error",
+    "no-lonely-if": "error",
+    "no-unneeded-ternary": "error",
     "no-use-before-define": ["error", "nofunc"],
+    "no-useless-return": "error",
+    "no-unused-vars": [
+      "error",
+      {
+        "ignoreRestSiblings": true
+      }
+    ],
```

Comment:
How about also setting the [`argsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#argsignorepattern) and [`caughtErrorsIgnorePattern`](https://eslint.org/docs/rules/no-unused-vars#caughterrorsignorepattern) options as follow?

```json
    "no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "caughtErrorsIgnorePattern": "^_"
        "ignoreRestSiblings": true
      }
    ],
```

Reply:
I don't think we currently follow this pattern. It seems to me we are trying more and more to be like xo; we might as well just use that config...

Reply:
<img width="551" alt="image" src="https://user-images.githubusercontent.com/473530/72420298-d16cb680-37c1-11ea-8b50-e0401c3a6427.png">

TypeScript's `noUnusedParameters` option allow `^_` pattern so I think  `argsIgnorePattern` and `caughtErrorsIgnorePattern` useful.

But it might be unnecessary now.